File: hooks/ding.php
#!/usr/bin/php
<?
require_once("boilerplate.php");

if(strlen($args) < 1) {
	reply_exit("Need a level");
}

$fullnick = mysql_escape_string($nick);
$nick = strtolower(substr($fullnick, 0, 3));
$level = $args;

db("delete from levels where nick like '$nick%'");
db("insert into levels(nick, level) values ('$fullnick', '$level');");

reply("ok");


?>

File: hooks/ding.rb
m = match_args(/(.+)/, '<level>')
if (account = Account.name_by_nick($msg.nick))
  DB[:levels].filter(:account => account).delete
  DB[:levels].insert(:account => account, :level => m[1])
  reply "account #{account} is #{m[1]}"
else
  DB[:levels].filter(:nick => $msg.nick).delete
  DB[:levels].insert(:nick => $msg.nick, :level => m[1])
  reply "#{$msg.nick} is #{m[1]}"
end
